### PR TITLE
Correct checklist title information

### LINKED ISSUE #212: inspec2ckl does not read the --metadata as JSON
Within PluginCLI, the read of the metadata file does not convert from JSON to a Hash. As a result all conversion downstream are not able to properly set attributes in the generated checklist when executing `inspec2ckl --metadata file`

Note the implementation here:

https://github.com/mitre/inspec_tools/blob/2862786fedefec4d488106bba8a9b580b95c74a6/lib/inspec_tools/plugin_cli.rb#L120-L125

### LINKED ISSUE #213: inspec2ckl does not create a title that matches original XCCDF document
When an XCCDF is used to generate a Checklist in STIGViewer the title output is different than when inspec2ckl is used to create the checklist title. An XCCDF attributes input is needed to allow for consistency with STIGViewer behavior.

The proposal here is to expose attributes in the same way as inspec2xccdf so that a matching title may be made in artifact creation.

https://github.com/mitre/inspec_tools/blob/2862786fedefec4d488106bba8a9b580b95c74a6/lib/inspec_tools/plugin_cli.rb#L58-L59

Updating inspec.rb with the following can then complete the update.

```ruby
    def generate_title(attributes, json)
      return "#{attributes['benchmark.title']} :: Version #{attributes['benchmark.version']}, #{attributes['benchmark.plaintext']}" if attributes['benchmark.title']
```

### PR DESCRIPTION
**Pull request by b-mad**
Add title conversion output and read host metadata correctly when producing checklist files.

Fixes #212, Fixes #213

### CODE REVIEW

File: examples/inspec2ckl/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+The InSpec to Checklist Parser scans and extracts the results defined in the Inspec JSON results and converts them into a
+Checklist XML file for use with STIGViewer.
+
+## XCCDF attributes YAML file
+
+### Inclusion of STIG information within the Checklist output.
+
+Inspec is unable to produce certain data that is required for conversion into a fully populated checklist. The following
+attributes supplement this data
```
Comment (Bialogs) [inline examples/inspec2ckl/README.md:11]: What are your thoughts about supporting generic SIData elements by looping over attributes vs. only supporting title, plaintext, version, date?

Comment (aaronlippold) [reply examples/inspec2ckl/README.md:11]: Hi all, I would like to resolve this. Any thoughts on where we want to go with this PR.

Thanks,

Aaron

File: lib/inspec_tools/inspec.rb [hunk 2]
```diff
@@ -220,9 +230,12 @@ def generate_ip
       ip
     end
 
-    def generate_title(title, json, date)
+    # @param attributes XCCDF attributes from source specification
+    def generate_title(attributes, json)
```
Comment (Bialogs) [inline lib/inspec_tools/inspec.rb:234]: The README needs to be updated about the behavior change when generating titles based on these changes.

File: lib/inspec_tools/inspec.rb [hunk 3]
```diff
@@ -220,9 +230,12 @@ def generate_ip
...
+      return "#{attributes['benchmark.title']} :: Version #{attributes['benchmark.version']}, #{attributes['benchmark.plaintext']}" if attributes['benchmark.title']
+
       title ||= "Untitled - Checklist Created from Automated InSpec Results JSON; Profiles: #{json['profiles'].map { |x| x['name'] }.join(' | ')}"
-      title + " Checklist Date: #{date || Date.today.to_s}"
+      title + " Checklist Date: #{Date.today}"
```
Comment (Bialogs) [inline lib/inspec_tools/inspec.rb:238]: This is a good change. From what I can tell date was never even used.

File: lib/inspec_tools/inspec.rb [hunk 4]
```diff
@@ -220,9 +230,12 @@ def generate_ip
       ip
     end
 
-    def generate_title(title, json, date)
+    # @param attributes XCCDF attributes from source specification
+    def generate_title(attributes, json)
+      return "#{attributes['benchmark.title']} :: Version #{attributes['benchmark.version']}, #{attributes['benchmark.plaintext']}" if attributes['benchmark.title']
```
Comment (Bialogs) [inline lib/inspec_tools/inspec.rb:235]: This logic needs more error checking for when plaintext and version are nil

**Review by Bialogs — CHANGES_REQUESTED** [review]

**Comment by Bialogs** [comment]
This needs to be rebased because of changes in #234

**Comment by Bialogs** [comment]
I am unable to push updates to your branch so I am continuing work https://github.com/mitre/inspec_tools/pull/246

**Pull request closed without merging**